renderer/FrameBuffer.cpp

```diff
@@ -789,16 +335,7 @@ if you absolutely must check for anything out of the ordinary, then do it here.
 ====================
 */
 void GL_Scissor( int x /* left*/, int y /* bottom */, int w, int h ) {
-	// x and y can be negative, but neither width nor height must be.
-	if ( w <= 0 || h <= 0 )
-		return;
-	if ( primaryOn ) {
-		x *= r_fboResolution.GetFloat();
-		y *= r_fboResolution.GetFloat();
-		w *= r_fboResolution.GetFloat();
-		h *= r_fboResolution.GetFloat();
-	}
-	qglScissor( x, y, w, h );
+	frameBuffers->activeFbo->SetScissorVidSize( x, y, w, h );
```

Comment:
I understand why scissor and viewport setters are non-static methods of FBO. But I think this sort-of implies that they are the part of FBO while they are not. When different FBO is bound, caller must reset viewport and scissor. This is easy to forget.

I wonder if it is possible to make scissor and viewport part of FBO state?... So that when new FBO is bound, it check if they are the same and send new values to GL driver if they have changed.

Reply:
You're right, it's not a well-designed interface. Hmm...

Having FBOs manage their individual viewport/scissor state is not the worst idea. The probability that you'd need to (re-)set these when switching FBOs is quite high. I still think a stack-based state management approach to allow localized state changes would catch more errors, but it's worth think about, either way.

Reply:
I just moved the functionality back into external helper functions GL_ViewportXXX and GL_ScissorXXX. In the end, there wasn't really any need to have them be member functions in the first placed.